pkg/controller/servicecatalog/provisionedproduct/utils.go

```diff
@@ -0,0 +1,101 @@
+package provisionedproduct
+
+import (
+	"fmt"
+
+	cfsdkv2types "github.com/aws/aws-sdk-go-v2/service/cloudformation/types"
+	svcsdk "github.com/aws/aws-sdk-go/service/servicecatalog"
+	"github.com/google/uuid"
+	"github.com/pkg/errors"
+	"k8s.io/utils/pointer"
+
+	svcapitypes "github.com/crossplane-contrib/provider-aws/apis/servicecatalog/v1alpha1"
+	aws "github.com/crossplane-contrib/provider-aws/pkg/clients"
+)
+
+func provisioningParamsAreChanged(cfStackParams []cfsdkv2types.Parameter, currentParams []*svcapitypes.ProvisioningParameter) bool {
+	if len(cfStackParams) != len(currentParams) {
+		return true
+	}
+
+	cfStackKeyValue := make(map[string]string)
+	for _, v := range cfStackParams {
+		cfStackKeyValue[*v.ParameterKey] = pointer.StringDeref(v.ParameterValue, "")
+	}
+
+	for _, v := range currentParams {
+		if cfv, ok := cfStackKeyValue[*v.Key]; ok && pointer.StringEqual(&cfv, v.Value) {
+			continue
+		} else {
+			return true
+		}
+	}
+
+	return false
+}
+
+// productOrArtifactAreChanged will attempt to determine whether or not the currently requested SC product and provisioning artifact IDs have changed when using names instead of IDs
+func (c *custom) productOrArtifactAreChanged(ds *svcapitypes.ProvisionedProductParameters, resp *svcsdk.ProvisionedProductDetail) (bool, error) { // nolint:gocyclo
```

Comment:
Could you please explain why you replaced `productOrArtifactAreNotChanged`(https://github.com/swisscom/provider-aws/pull/1/conflicts) to ` productOrArtifactAreChanged`, am asking not about `Not` : ) It's okay. But about meaning...
```suggestion
func (c *custom) productOrArtifactAreChanged(ds *svcapitypes.ProvisionedProductParameters, resp *svcsdk.ProvisionedProductDetail) (bool, error) {
	// ProvisioningArtifactID and ProvisioningArtifactName are mutual exclusive params, the same about ProductID and ProductName
	// But if describe a provisioned product aws api will return only IDs, so it's impossible to compare names with ids
	// Conditional statement below works only if desired state includes ProvisioningArtifactID and ProductID
	if ds.ProvisioningArtifactID != nil && ds.ProductID != nil &&
		(*ds.ProvisioningArtifactID != *resp.ProvisioningArtifactId ||
			*ds.ProductID != *resp.ProductId) {
		return true, nil
		// In case if desired state includes not only IDs provider runs func `getArtifactID`, which produces
		// additional request to aws api and retrieves an artifact id based on ProductId/ProductName and ProvisioningArtifactId/ProvisioningArtifactName
		// for further comparison with artifact id in the current state
	} else if ds.ProvisioningArtifactName != nil || ds.ProductName != nil {
		desiredArtifactID, err := c.getArtifactID(ds)
		if err != nil {
			return true, err
		}
		if desiredArtifactID != *resp.ProvisioningArtifactId {
			return true, nil
		}
	}
	return false, nil
```